hepcrawl/parsers/elsevier.py

```diff
@@ -372,6 +372,16 @@ def page_end(self):
         ).extract_first()
         return page_end
 
+    @property
+    def imprints_date(self):
+        imprints_date = self.root.xpath(
+            "string(./RDF/Description/availableOnlineInformation/availableOnline)"
+        ).extract_first()
+        if imprints_date:
+            return PartialDate.parse(imprints_date).dumps()
+        if self.publication_date:
+            return self.publication_date.dumps()
```

Comment:
it's ok, I would just move the logic with the `publication_date` to the `parse` since `imprints_date` should handle only one thing

Reply:
Yes, but with a cost of a lower readability imo. But I can move it there.